search/grammar_v2_engine.go

```diff
@@ -399,3 +487,61 @@ func (e *ESEngine) collectionHitFromSql(query string) (*elastic.SearchHit, error
 	}
 	return hit, nil
 }
+
+func retrieveTextVarValues(str string) []string {
+	runes := []rune(str)
+	var filtered []rune
+	var textVarValues []string
+	var inHighlight bool
+	for i, r := range runes {
+		if r == PERCULATE_HIGHLIGHT_SEPERATOR || i == len(runes)-1 {
+			inHighlight = !inHighlight
+			if inHighlight && len(filtered) > 0 {
+				if r != PERCULATE_HIGHLIGHT_SEPERATOR {
+					filtered = append(filtered, r)
+				}
+				trimmed := strings.Trim(string(filtered), " ")
+				if trimmed != "" {
+					textVarValues = append(textVarValues, trimmed)
+				}
+			}
+			filtered = make([]rune, 0)
+		} else if !inHighlight {
+			filtered = append(filtered, r)
+		}
+	}
+	return textVarValues
+}
+
+func (e *ESEngine) filterSearch(requests []*elastic.SearchRequest) ([]*elastic.SearchResult, map[string]bool, *float64, error) {
```

Comment:
What that do?

Reply:
Results search according to content type filter. This logic is separated from the SearchGrammarsV2 function to minimize the size of SearchGrammarsV2  function .

Reply:
Add comment.

Reply:
Added